custom/icds_reports/utils/aggregation_helpers/agg_awc.py

```diff
@@ -609,3 +609,67 @@ def _transform_column(column_tuple):
             group_by=", ".join(group_by),
             child_is_test=child_location
         )
+
+    def weekly_updates(self):
+        yield """
+        UPDATE  agg_awc SET
+            usage_num_hh_reg = ut.usage_num_hh_reg,
+            is_launched = ut.is_launched,
+            num_launched_states = ut.num_launched_awcs,
+            num_launched_districts = ut.num_launched_awcs,
+            num_launched_blocks = ut.num_launched_awcs,
+            num_launched_supervisors = ut.num_launched_awcs,
+            num_launched_awcs = ut.num_launched_awcs,
+            usage_awc_num_active = ut.usage_awc_num_active
+        FROM (
+        SELECT
+            awc_id,
+            month,
+            sum(add_household) AS usage_num_hh_reg,
+            CASE WHEN sum(add_household) > 0 THEN 'yes' ELSE 'no' END as is_launched
+            CASE WHEN sum(add_household) > 0 THEN 1 ELSE 0 END as num_launched_awcs
+            CASE WHEN (sum(due_list_ccs) + sum(due_list_child) + sum(pse) + sum(gmp) + sum(thr) + sum(home_visit) + sum(add_pregnancy) + sum(add_household)) >= 15 THEN 1 ELSE 0 END AS usage_awc_num_active
+            FROM "{usage_table}"
+            WHERE month >= %(start_date)s GROUP BY awc_id, month
+        ) ut
+        WHERE ut.month <= agg_awc.month AND ut.awc_id = agg_awc.awc_id AND agg_awc.aggregation_level=5
+        AND agg_awc.num_launched_awcs = 0 AND ut.num_launched_awcs != 0;
+        """.format(
+            usage_table=self._ucr_tablename('static-usage_forms')
+        ), {
+            'start_date': self.month_start_6m
+        }
+
+        yield """
+        UPDATE  agg_awc SET
+            usage_num_hh_reg = ut.sum_usage_num_hh_reg,
+            num_launched_states = ut.num_launched_supervisors,
+            num_launched_districts = ut.num_launched_supervisors,
+            num_launched_blocks = ut.num_launched_supervisors,
+            num_launched_supervisors = ut.num_launched_supervisors,
+            num_launched_awcs = ut.sum_num_launched_awcs,
+            usage_awc_num_active = ut.sum_usage_awc_num_active
+        FROM (
+            SELECT
+                state_id,
+                CASE WHEN GROUPING(district_id) = 1 THEN 'All' else district_id END,
+                CASE WHEN GROUPING(block_id) = 1 THEN 'All' else block_id END,
+                CASE WHEN GROUPING(supervisor_id) = 1 THEN 'All' else supervisor_id END,
```

Comment:
Thoughts on this approach vs `COALESCE(supervisor_id, 'All')` ? I used that in AAA (though thinking about leaving it null in the future). I do like that yours makes it a bit more explicit why we need to have this.

I'm undecided on which approach to take, but I think we should keep it consistent. I can follow up on the AAA SQL if we think this is a better approach

Reply:
I prefer this approach since it would prevent grouping any data where the location field is actually null in the data (even if that shouldn't be possible). Also as you said it's more explicit.